sys/kern/kern_dbquery.c

```diff
@@ -22,8 +22,7 @@ int sys_dbquery (struct thread *td, struct dbquery_args *uap) {
         return ENOMEM;
     }
 
-    sprintf(kernel_buffer, "Not implemented");
-    kernel_buffer[499] = '\0';
+    sprintf(kernel_buffer, "Not implemented\n");
```

Comment:
@stelkasouridis : Shouldn't we not write anything into the buffer, and indicate an error with the integer return value from the function? I guess the return indicates the amount of data we've written back, so if we return 0 to the caller, it would mean that we haven't written anything to this buffer.